dlt/extract/hints.py

```diff
@@ -440,9 +441,14 @@ def _merge_merge_disposition_dict(dict_: Dict[str, Any]) -> None:
 
         mddict: TMergeDispositionDict = deepcopy(dict_["write_disposition"])
         if mddict is not None:
-            dict_["x-merge-strategy"] = (
-                mddict["strategy"] if "strategy" in mddict else DEFAULT_MERGE_STRATEGY
-            )
+            dict_["x-merge-strategy"] = DEFAULT_MERGE_STRATEGY
+            if "strategy" in mddict:
+                if mddict["strategy"] not in MERGE_STRATEGIES:
+                    raise ValueError(
```

Comment:
Is this the right way/place to do user input validation?

Reply:
Moved it to `_set_hints` in https://github.com/dlt-hub/dlt/pull/1466/commits/479e52adacd7b504123bf8b2b7bb74f802c63c68